front/lib/swr/data_source_documents.ts

```diff
@@ -0,0 +1,58 @@
+import type { DataSourceViewType, LightWorkspaceType } from "@dust-tt/types";
+import type {
+  PatchDataSourceWithNameDocumentRequestBody,
+  PostDataSourceWithNameDocumentRequestBody,
+} from "@dust-tt/types";
+import type { SWRMutationConfiguration } from "swr/mutation";
+import useSWRMutation from "swr/mutation";
+
+import { fetcherWithBody } from "@app/lib/swr/swr";
+import type { PostDocumentResponseBody } from "@app/pages/api/w/[wId]/spaces/[spaceId]/data_sources/[dsId]/documents";
+import type { PatchDocumentResponseBody } from "@app/pages/api/w/[wId]/spaces/[spaceId]/data_sources/[dsId]/documents/[documentId]";
+
+async function sendPatchRequest(
+  url: string,
+  {
+    arg,
+  }: {
+    arg: {
+      documentBody: PatchDataSourceWithNameDocumentRequestBody;
+    };
+  }
+) {
+  const res = await fetcherWithBody([url, arg.documentBody, "PATCH"]);
+  return res;
+}
```

Comment:
`nit` but probably easier to read:
```suggestion
async function sendPatchRequest(
  url: string,
  { documentBody }: { documentBody: PatchDataSourceWithNameDocumentRequestBody }
) {
  return fetcherWithBody([url, documentBody, "PATCH"]);
}
```

Reply:
Agreed but I think we don't have that flexibility according to https://swr.vercel.app/docs/mutation#useswrmutation-parameters

Reply:
Ah yes missed the point that it was passed directly to SWR 🙈.